fix(api-cors-lambda-crud-dynamodb example): Implement use of NodejsFunction, replaced deprecated API

ryparker opened the pull request:
## Summary

This PR changes the following:

- Implement use of NodejsFunction.
- Upgraded lambda runtime from Node 10 -> 14.
- Renamed `/src/` -> `/lambdas/`. This was a personal choice but I believe it makes more sense to label this as anything but `src` since it does not share the same node_modules as the root `package.json`.
- Created `/lambdas/package.json` to define modules required by the lambda functions.
- Added `"postinstall"` script to root package.json which installs the nested `/lambdas/package.json` dependencies after running `npm install` from the root.


Fixes #276, #468, #195
Would have pushed my fixes to #278 however the PR author's branch is protected. 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/index.ts:
```diff
@@ -14,62 +16,94 @@ export class ApiLambdaCrudDynamoDBStack extends cdk.Stack {
-      code: new lambda.AssetCode('src'),
-      handler: 'get-one.handler',
-      runtime: lambda.Runtime.NODEJS_10_X,
+    const getOneLambda = new NodejsFunction(this, 'getOneItemFunction', {
```
If you are going to use direct component imports (ie. `import { NodejsFunction } from '...';` vs `import * as lambdanode from '...';` and `new NodejsFunction()` vs `new lambdanode.NodejsFunction()`) please do so for all other components (`lambda.Runtime`, `dynamodb.Table`, etc.)

NGL321 replied:
Just to clarify, I think importing components directly is a good approach for examples as it looks cleaner and is considered good practice for projects with few imports.

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/index.ts:
```diff
@@ -14,62 +16,94 @@ export class ApiLambdaCrudDynamoDBStack extends cdk.Stack {
-      handler: 'update-one.handler',
-      runtime: lambda.Runtime.NODEJS_10_X,
+    const updateOne = new NodejsFunction(this, 'updateItemFunction', {
+      entry: join(__dirname, 'lambdas', 'update-one.ts'), // accepts .js, .jsx, .ts and .tsx files
```
Imo these comments only need to appear on the first occurrence (unless you have strong feelings to the contrary).

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/lambdas/create.ts:
```diff
@@ -1,13 +1,15 @@
-const AWS = require('aws-sdk');
-const db = new AWS.DynamoDB.DocumentClient();
-const uuidv4 = require('uuid/v4');
+import * as AWS from 'aws-sdk';
```
While it contradicts the above comment, I think non-specific imports for the AWS SDK makes sense even if it is contradictory since the SDK is specifically designed for use with general imports.

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/lambdas/tsconfig.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../tsconfig.json"
+}
```
New line needed

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/lambdas/update-one.ts:
```diff
@@ -1,12 +1,14 @@
   DYNAMODB_EXECUTION_ERROR = `Error: Execution update, caused a Dynamodb error, please take a look at your CloudWatch Logs.`;
 
-export const handler = async (event: any = {}) : Promise <any> => {
+const db = new AWS.DynamoDB.DocumentClient();
```
Imo should be separated by line break as was done in previous files.

NGL321 commented on typescript/api-cors-lambda-crud-dynamodb/tsconfig.json:
```diff
@@ -15,7 +19,9 @@
+    "include": [
+        "*.ts"
+    ]
+}
```
Another newline

NGL321 reviewed (changes requested):
Looks really good. I have a couple notes for improvements, then its ready to merge

ryparker commented:
> Looks really good. I have a couple notes for improvements, then its ready to merge

Thanks for the suggestions. I implemented all of them and I went a bit further and cleaned up the index.ts file. Specifically the lambda props. Instead of duplicating them in each lambda declaration I stored them in `nodeJsFunctionProps` and reused them in each lambda declaration.

LMK what you think.

ryparker commented on typescript/api-cors-lambda-crud-dynamodb/lambdas/package.json:
```diff
@@ -0,0 +1,15 @@
+    "aws-sdk": "*",
+    "uuid": "*"
+  }
+}
```
Strange this should have a newline....

ryparker replied:
For some reason I cannot see the newline in GitHub's review UI. However when I click "edit file" I see there is already a new line. LMK if you don't see it during your review @NGL321.

ryparker replied:
Did some digging into this and it looks like every file does have a newline char added however you'll only see it in your local editor or when editing the file via GitHub's "edit file" UI. If you want the newline to render in GitHubs "view file" view, then you'll have to add 2 newlines to the bottom of the file.

Is this alright as is? @NGL321

NGL321 reviewed (approved):
Looks great!! 🚀
